pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -2599,6 +2648,97 @@ func TestContextInvalid(t *testing.T) {
 			Also(apis.ErrGeneric(`non-existent variable in "$(context.pipeline.missing-foo)"`, "value")).
 			Also(apis.ErrGeneric(`non-existent variable in "$(context.pipelineRun.missing-foo)"`, "value")).
 			Also(apis.ErrGeneric(`non-existent variable in "$(context.pipelineTask.missing-foo)"`, "value")),
+	}, {
+		name: "invalid string context variable for pipeline in include matrix",
+		tasks: []PipelineTask{{
+			Name:    "bar",
+			TaskRef: &TaskRef{Name: "bar-task"},
+			Params: []Param{{
+				Name: "a-param", Value: ParamValue{StringVal: "$(context.pipeline.missing)"},
+			}},
+			Matrix: &Matrix{
+				Include: []MatrixInclude{{
+					Name: "build-1",
+					Params: []Param{{
+						Name: "a-param-foo", Value: ParamValue{Type: ParamTypeString, StringVal: "$(context.pipeline.missing-foo)"}}},
+				}}},
+		}},
+		expectedError: *apis.ErrGeneric("").Also(&apis.FieldError{
+			Message: `non-existent variable in "$(context.pipeline.missing)"`,
+			Paths:   []string{"value"},
+		}).Also(&apis.FieldError{
+			Message: `non-existent variable in "$(context.pipeline.missing-foo)"`,
+			Paths:   []string{"value"},
+		}),
+	}, {
+		name: "invalid string context variable for pipelineRun",
+		tasks: []PipelineTask{{
+			Name:    "bar",
+			TaskRef: &TaskRef{Name: "bar-task"},
+			Params: []Param{{
+				Name: "a-param", Value: ParamValue{StringVal: "$(context.pipelineRun.missing)"},
+			}},
+			Matrix: &Matrix{
+				Include: []MatrixInclude{{
+					Name: "build-1",
+					Params: []Param{{
+						Name: "a-param-foo", Value: ParamValue{Type: ParamTypeString, StringVal: "$(context.pipelineRun.missing-foo)"}}},
+				}}},
+		}},
+		expectedError: *apis.ErrGeneric("").Also(&apis.FieldError{
+			Message: `non-existent variable in "$(context.pipelineRun.missing)"`,
+			Paths:   []string{"value"},
+		}).Also(&apis.FieldError{
+			Message: `non-existent variable in "$(context.pipelineRun.missing-foo)"`,
+			Paths:   []string{"value"},
+		}),
+	}, {
+		name: "invalid string context variable for pipelineTask",
+		tasks: []PipelineTask{{
+			Name:    "bar",
+			TaskRef: &TaskRef{Name: "bar-task"},
+			Params: []Param{{
+				Name: "a-param", Value: ParamValue{StringVal: "$(context.pipelineTask.missing)"},
+			}},
+			Matrix: &Matrix{
+				Include: []MatrixInclude{{
+					Name: "build-1",
+					Params: []Param{{
+						Name: "a-param-foo", Value: ParamValue{Type: ParamTypeString, StringVal: "$(context.pipelineTask.missing-foo)"}}},
+				}}},
+		}},
+		expectedError: *apis.ErrGeneric("").Also(&apis.FieldError{
+			Message: `non-existent variable in "$(context.pipelineTask.missing)"`,
+			Paths:   []string{"value"},
+		}).Also(&apis.FieldError{
+			Message: `non-existent variable in "$(context.pipelineTask.missing-foo)"`,
+			Paths:   []string{"value"},
+		}),
+	}, {
+		name: "invalid array context variables for pipeline, pipelineTask and pipelineRun",
```

Comment:
what is this test case designed to test for matrix.include that isn't covered by the other test cases?

Reply:
It was copying over an existing test with combination for Matrix.Include params. Deleted though